src/PlusDataCollection/RevopointSurface/vtkPlusRevopointSurface.cxx

```diff
@@ -0,0 +1,385 @@
+/*=Plus=header=begin======================================================
+Program: Plus
+Copyright (c) Laboratory for Percutaneous Surgery. All rights reserved.
+See License.txt for details.
+=========================================================Plus=header=end*/
+
+#include "PlusConfigure.h"
+#include "vtkPlusRevopointSurface.h"
+
+// Local includes
+#include "vtkPlusChannel.h"
+#include "vtkPlusDataSource.h"
+
+// VTK includes
+#include <vtkImageData.h>
+#include <vtkObjectFactory.h>
+
+// Revopoint SDK includes
+#include "3DCamera.hpp"
+
+// Stl includes
+#include <vector>
+#include <string>
+#include <memory>
+#include <algorithm>
+
+//----------------------------------------------------------------------------
+vtkStandardNewMacro(vtkPlusRevopointSurface);
+
+namespace
+{
+  struct SurfaceStreamConfig
+  {
+    std::string Id;
+    StreamInfo StreamProps;
+    int DepthRange[2];
+    Intrinsics Intr;
+    Extrinsics Extr;
+    vtkPlusDataSource* Source{nullptr};
+  };
+
+  std::string convertErrorCode(ERROR_CODE error)
+  {
+    switch (error)
+    {
+      case ERROR_CODE::ERROR_PARAM:
+        return "Bad param";
+      case ERROR_CODE::ERROR_DEVICE_NOT_FOUND:
+        return "Device not found";
+      case ERROR_CODE::ERROR_DEVICE_NOT_CONNECT:
+        return "Device not connected";
+      case ERROR_CODE::ERROR_DEVICE_BUSY:
+        return "Device busy";
+      case ERROR_CODE::ERROR_STREAM_NOT_START:
+        return "Device not started";
+      case ERROR_CODE::ERROR_STREAM_BUSY:
+        return "Stream busy";
+      case ERROR_CODE::ERROR_FRAME_TIMEOUT:
+        return "Frame timeout";
+      case ERROR_CODE::ERROR_NOT_SUPPORT:
+        return "Not supported";
+      case ERROR_CODE::ERROR_PROPERTY_GET_FAILED:
+        return "Getting property failed";
+      case ERROR_CODE::ERROR_PROPERTY_SET_FAILED:
+        return "Setting property failed";
+      default:
+        break;
+    }
+    return "Success";
+  }
+}
+
+//----------------------------------------------------------------------------
+class vtkPlusRevopointSurface::vtkInternal
+{
+public:
+  vtkPlusRevopointSurface* External;
+  SurfaceStreamConfig DepthStream;
+  cs::ICameraPtr Camera;
+  cs::IFramePtr DepthFrame;
+
+  vtkInternal(vtkPlusRevopointSurface* external)
+    : External(external)
+  {
+    Camera = cs::getCameraPtr();
```

Comment:
Use "this->" when accessing member variables/functions in vtk* files. ([VTK coding standard](https://docs.google.com/document/d/1nzinw-dR5JQRNi_gb8qwLL5PnkGMK2FETlQGLr10tZw/edit#heading=h.z72938ebor6h)).